sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs

```diff
@@ -75,8 +75,8 @@ public void Initialize(ExtensionConfigContext context)
                 .AddConverter<JToken, string>((jtoken) => jtoken.ToString(Formatting.Indented))
                 .AddConverter<JToken, string[]>((jarray) => jarray.Select(ar => ar.ToString(Formatting.Indented)).ToArray())
                 .AddConverter<JToken, DirectInvokeString>((jtoken) => new DirectInvokeString(null))
-                .AddConverter<JToken, EventGridEvent>((jobject) => jobject.ToObject<EventGridEvent>()) // surface the type to function runtime
-                .AddConverter<JToken, EventGridEvent[]>((jobject) => jobject.ToObject<EventGridEvent[]>()) // surface the type to function runtime
+                .AddConverter<JToken, EventGridEvent>((jobject) => EventGridEvent.Parse(jobject.ToString()).Single()) // surface the type to function runtime
```

Comment:
This needed to be changed because BinaryData cannot be constructed with a JToken. This was an issue before, but by making Data a property rather than having the GetData method, an exception was thrown when the EventGridEvent ctor is used by NewtonSoft.